### PR TITLE
Add Manifest fixtures to test suite

### PR DESCRIPTION
**Pull request by colinwahl**
This PR adds fixtures for the `Manifest` type, making it easier to create a `Manifest` for testing.

In particular, this is meant to replace the painful experience of hand-writing a `Manifest` yourself for testing. See the deleted `Test.Support.Manifest` to see how bad it used to be :)

Packaged with the manifest are some utilities for setting the name/dependencies/version/description. These can be used to get fine grained control over just the fields from the manifest that you'd like to change - see the new implementation in `Test.Registry.Index` for how it could be used in action.

Alongside those utilities is a utility that take a depth argument and produces a perfect binary tree of manifests, where the `Manifest` at index 0 is the root of the tree, and all transitive dependencies are contained within the tree. As part of uploading to pursuit we'd like to verify that the build plan provided by the user fulfills all transitive dependencies of a package, and this utility will be useful for generating fake manifests to test said procedure.

I also had to fix one bug in `Registry.Schema` - we had forgotten to `encode` the `files` field, meaning that `Manifest` encoding/decoding did not round trip! Sadly, our existing round trip tests did not include a `Manifest` that had the `files` field filled in, so I replaced them with a test that actually includes all fields so we can ensure they round trip.

### CODE REVIEW

File: ci/test/Fixture/Manifest.purs [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  fixture = unsafeFromRight $ Version.parseVersion Version.Strict "1.0.0"
+
+instance Fixture License where
+  fixture = unsafeFromRight $ SPDX.parse "MIT"
+
+instance Fixture Owner where
+  fixture = Owner
+    { email: fixture
+    , keytype: fixture
+    , public: fixture
```
Comment (thomashoneyman) [inline ci/test/Fixture/Manifest.purs]: Can we put real (dummy) values in here, such as:

```suggestion
    { email: "user@example.com"
    , keytype: "ssh-ed25519"
    , public: "AAAAC3NzaC1lZDI1NTE5AAAAIF6B6R9yp8yFM3wYcOhGO0EyZAefKWkvsTET+KeaegY/"
    }
```

File: ci/test/Fixture/Manifest.purs [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    }
+
+instance Fixture Location where
+  fixture = GitHub { owner: fixture, repo: fixture, subdir: Nothing }
```
Comment (thomashoneyman) [inline ci/test/Fixture/Manifest.purs:43]: Same here for some kind of dummy values, like `"owner"` and `"repo"`

File: ci/test/Fixture/Manifest.purs (hunk 1 above)
Comment (thomashoneyman) [reply ci/test/Fixture/Manifest.purs]: The idea being that the fixtures are by default _valid_, even if they're dummy values.

**Review by thomashoneyman — COMMENTED** [review]
Other than putting in real default values for the fixtures, I'm quite happy with this! Thanks!

File: ci/test/Fixture/Manifest.purs (hunk 1 above)
Comment (colinwahl) [reply ci/test/Fixture/Manifest.purs]: Yes - that makes sense!

**Review by f-f — APPROVED** [review]
Looks great!

**Comment by colinwahl** [comment]
This intends to fix #373 - forgot to link the issue!

**Pull request merged**